src/element.rs

```diff
@@ -116,13 +117,35 @@ impl IconElement {
         match icon_type.encoding() {
             #[cfg(feature = "pngio")]
             Encoding::JP2PNG => {
+                let image: Image;
                 if self.data.starts_with(&JPEG_2000_FILE_MAGIC_NUMBER) {
-                    let msg = "element to be decoded contains JPEG 2000 \
-                               data, which is not yet supported";
-                    return Err(Error::new(ErrorKind::InvalidInput, msg));
+                    let bytes = &self.data[12..];
+                    let codec = jp2k::Codec::jp2();
+                    let stream = jp2k::Stream::from_bytes(bytes).unwrap();
+
+                    let jp2k::ImageBuffer { buffer, width, height, num_bands } = jp2k::ImageBuffer::build(
+                        codec,
+                        stream,
+                        jp2k::DecodeParams::default().with_reduce_factor(1),
+                    )
+                    .unwrap();
+
+                    let image_rip = rips::Image::from_memory(
+                        buffer,
+                        width as i32,
+                        height as i32,
+                        num_bands as i32,
+                        rips::VipsBandFormat::VIPS_FORMAT_UCHAR,
+                    )
+                    .unwrap();
+                    // Rip image to imagem
+                   image = Image::read_png(io::Cursor::new(image_rip.to_bytes())).unwrap()
```

Comment:
Does the test currently pass?  I'm confused by this line because it doesn't seem like `rips::Image::to_bytes()` returns PNG data?  Or does it?